### PR TITLE
docker buildx: use cache and cross compile

### PR DESCRIPTION
**Pull request by 6543**
https://docs.docker.com/build/guide/multi-platform/#platform-build-arguments

### CODE REVIEW

**Review by mstruebing — COMMENTED** [review]
Hey thanks for your fix - Just a small correction/question otherwise this looks good to me.

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,12 @@
-FROM golang:1.19-alpine as build
+FROM --platform=$BUILDPLATFORM golang:1.19-alpine as build
```
[the remark this answers is not in the document]
Comment (6543) [reply Dockerfile:1]: no this tells to run with host arch/system independent of the target arch/system. so the first stage is no emulation but it run native and cross compile, witch go supports natively (without any issue as long as CGO is disabled)

the second stage dont have this and so it gets emulated. emulation do have a noticeable overhead but if we dont do CPU rich tasks it's fine (and you dont have a choice anyway in this case)

File: Dockerfile [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
 
 RUN apk add --no-cache git
 WORKDIR /ec
 COPY . /ec
-RUN GO111MODULE=on CGO_ENABLED=0 go build -ldflags "-X main.version=$(cat VERSION | tr -d '\n')" -o bin/ec ./cmd/editorconfig-checker/main.go
+ARG TARGETOS TARGETARCH
```
Comment (6543) [inline Dockerfile:6]: https://docs.docker.com/build/guide/multi-platform/#platform-build-arguments

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (6543) [reply Dockerfile:1]: PS: I would call it enhancement not fix, as it (only) improve multi arch builds

**Review by mstruebing — APPROVED** [review]
Thanks!

**Pull request merged**